### ISSUE TITLE
[Phase 5] Implement BreakoutStrategy

### ISSUE THREAD

**Issue by kimrejstrom** [kimrejstrom/alpacalyzer-algo-trader#26]
## Overview

Implement a `BreakoutStrategy` that identifies and trades price breakouts from consolidation patterns.

## Background

Breakout trading is a popular strategy that enters positions when price breaks through support/resistance levels with volume confirmation. This provides an alternative to the momentum strategy.

## Strategy Logic

Entry conditions:
1. Price breaks above resistance or below support
2. Volume spike confirms breakout (> 1.5x average)
3. ATR indicates sufficient volatility
4. No recent false breakouts

Exit conditions:
1. Target reached (based on pattern height)
2. Stop loss triggered
3. Breakout fails (price returns to consolidation)

## Tasks

- [ ] Create `src/alpacalyzer/strategies/breakout.py`
- [ ] Implement `BreakoutStrategy` class
- [ ] Add breakout detection logic
- [ ] Create support/resistance identification
- [ ] Add volume confirmation
- [ ] Write unit tests

## Implementation

```python
# src/alpacalyzer/strategies/breakout.py

from dataclasses import dataclass
from datetime import datetime, UTC, timedelta
from typing import Literal

import pandas as pd
import numpy as np

from alpacalyzer.strategies.base import Strategy, StrategyConfig, EntrySignal, ExitSignal


@dataclass
class BreakoutConfig(StrategyConfig):
    """Configuration for breakout strategy."""
    name: str = "breakout"
    
    # Consolidation detection
    consolidation_periods: int = 20  # Bars to look for consolidation
    consolidation_range_pct: float = 0.05  # Max range % for consolidation
    
    # Breakout confirmation
    min_volume_ratio: float = 1.5  # Min volume vs average
    breakout_buffer_pct: float = 0.002  # Buffer above/below level
    
    # Position sizing
    risk_pct_per_trade: float = 0.02
    target_multiple: float = 2.0  # Target as multiple of risk
    
    # Filters
    min_atr: float = 0.5  # Minimum ATR in dollars
    max_false_breakouts: int = 2  # Max recent false breakouts


class BreakoutStrategy(Strategy):
    """
    Breakout trading strategy.
    
    Identifies consolidation patterns and enters when price
    breaks out with volume confirmation.
    """
    
    def __init__(self, config: BreakoutConfig | None = None):
        self.config = config or BreakoutConfig()
        self._false_breakout_count: dict[str, int] = {}
    
    @property
    def name(self) -> str:
        return self.config.name
    
    def should_enter(
        self,
        ticker: str,
        price: float,
        df: pd.DataFrame,
    ) -> tuple[bool, EntrySignal | None]:
        """Check if breakout conditions are met."""
        
        if len(df) < self.config.consolidation_periods + 10:
            return False, None
        
        # Get recent data
        recent = df.tail(self.config.consolidation_periods)
        latest = df.iloc[-1]
        
        # Calculate support/resistance
        resistance = recent["high"].max()
        support = recent["low"].min()
        range_pct = (resistance - support) / support
        
        # Check if in consolidation
        if range_pct > self.config.consolidation_range_pct:
            return False, None
        
        # Check volume
        avg_volume = df["volume"].tail(50).mean()
        current_volume = latest["volume"]
        volume_ratio = current_volume / avg_volume if avg_volume > 0 else 0
        
        if volume_ratio < self.config.min_volume_ratio:
            return False, None
        
        # Check ATR
        atr = self._calculate_atr(df)
        if atr < self.config.min_atr:
            return False, None
        
        # Check for false breakouts
        if self._false_breakout_count.get(ticker, 0) >= self.config.max_false_breakouts:
            return False, None
        
        # Check breakout direction
        buffer = price * self.config.breakout_buffer_pct
        
        # Bullish breakout
        if price > resistance + buffer:
            stop_loss = support - atr
            target = price + (price - support) * self.config.target_multiple
            
            return True, EntrySignal(
                ticker=ticker,
                side="long",
                price=price,
                stop_loss=stop_loss,
                target=target,
                confidence=self._calculate_confidence(df, "bullish"),
                reason=f"Bullish breakout above {resistance:.2f} with {volume_ratio:.1f}x volume",
            )
        
        # Bearish breakout
        if price < support - buffer:
            stop_loss = resistance + atr
            target = price - (resistance - price) * self.config.target_multiple
            
            return True, EntrySignal(
                ticker=ticker,
                side="short",
                price=price,
                stop_loss=stop_loss,
                target=target,
                confidence=self._calculate_confidence(df, "bearish"),
                reason=f"Bearish breakout below {support:.2f} with {volume_ratio:.1f}x volume",
            )
        
        return False, None
    
    def should_exit(
        self,
        ticker: str,
        entry_price: float,
        current_price: float,
        position_side: str,
        df: pd.DataFrame,
        **kwargs,
    ) -> tuple[bool, ExitSignal | None]:
        """Check if position should be exited."""
        
        stop_loss = kwargs.get("stop_loss", 0)
        target = kwargs.get("target", float("inf"))
        
        # Stop loss
        if position_side == "long" and current_price <= stop_loss:
            self._record_false_breakout(ticker)
            return True, ExitSignal(
                ticker=ticker,
                side=position_side,
                price=current_price,
                reason="stop_loss",
                urgency="immediate",
            )
        
        if position_side == "short" and current_price >= stop_loss:
            self._record_false_breakout(ticker)
            return True, ExitSignal(
                ticker=ticker,
                side=position_side,
                price=current_price,
                reason="stop_loss",
                urgency="immediate",
            )
        
        # Target reached
        if position_side == "long" and current_price >= target:
            self._clear_false_breakouts(ticker)
            return True, ExitSignal(
                ticker=ticker,
                side=position_side,
                price=current_price,
                reason="target_reached",
                urgency="normal",
            )
        
        if position_side == "short" and current_price <= target:
            self._clear_false_breakouts(ticker)
            return True, ExitSignal(
                ticker=ticker,
                side=position_side,
                price=current_price,
                reason="target_reached",
                urgency="normal",
            )
        
        # Check for failed breakout (price returned to range)
        recent = df.tail(self.config.consolidation_periods)
        resistance = recent["high"].max()
        support = recent["low"].min()
        
        if position_side == "long" and current_price < resistance:
            # Failed bullish breakout
            return True, ExitSignal(
                ticker=ticker,
                side=position_side,
                price=current_price,
                reason="breakout_failed",
                urgency="urgent",
            )
        
        if position_side == "short" and current_price > support:
            # Failed bearish breakout
            return True, ExitSignal(
                ticker=ticker,
                side=position_side,
                price=current_price,
                reason="breakout_failed",
                urgency="urgent",
            )
        
        return False, None
    
    def _calculate_atr(self, df: pd.DataFrame, period: int = 14) -> float:
        """Calculate Average True Range."""
        high = df["high"]
        low = df["low"]
        close = df["close"]
        
        tr1 = high - low
        tr2 = abs(high - close.shift())
        tr3 = abs(low - close.shift())
        
        tr = pd.concat([tr1, tr2, tr3], axis=1).max(axis=1)
        return tr.tail(period).mean()
    
    def _calculate_confidence(self, df: pd.DataFrame, direction: str) -> float:
        """Calculate confidence score for the breakout."""
        confidence = 50.0
        
        # Volume factor
        avg_volume = df["volume"].tail(50).mean()
        current_volume = df.iloc[-1]["volume"]
        if current_volume > avg_volume * 2:
            confidence += 15
        elif current_volume > avg_volume * 1.5:
            confidence += 10
        
        # Trend alignment
        sma_50 = df["close"].tail(50).mean()
        sma_20 = df["close"].tail(20).mean()
        price = df.iloc[-1]["close"]
        
        if direction == "bullish" and price > sma_20 > sma_50:
            confidence += 15
        elif direction == "bearish" and price < sma_20 < sma_50:
            confidence += 15
        
        # Range tightness (tighter = better)
        recent = df.tail(self.config.consolidation_periods)
        range_pct = (recent["high"].max() - recent["low"].min()) / recent["low"].min()
        if range_pct < 0.03:
            confidence += 10
        
        return min(confidence, 95.0)
    
    def _record_false_breakout(self, ticker: str) -> None:
        """Record a false breakout for a ticker."""
        self._false_breakout_count[ticker] = self._false_breakout_count.get(ticker, 0) + 1
    
    def _clear_false_breakouts(self, ticker: str) -> None:
        """Clear false breakout count after successful trade."""
        self._false_breakout_count[ticker] = 0
```

## Acceptance Criteria

- [ ] `BreakoutStrategy` class implements `Strategy` protocol
- [ ] `BreakoutConfig` with all parameters
- [ ] Support/resistance identification
- [ ] Volume confirmation logic
- [ ] False breakout tracking
- [ ] Configurable risk/reward
- [ ] Unit tests for all conditions
- [ ] Backtest performance validation

## Dependencies

- #5 Strategy Protocol
- #6 Strategy Registry
- #25 Backtesting Framework

## Estimated Effort

3-4 hours
